src/test/kotlin/org/opensearch/replication/repository/RemoteClusterRepositoriesServiceTests.kt

```diff
@@ -0,0 +1,49 @@
+package org.opensearch.replication.repository
+
+import com.carrotsearch.randomizedtesting.annotations.ThreadLeakScope
+import org.mockito.Mockito
+import org.opensearch.Version
+import org.opensearch.cluster.node.DiscoveryNode
+import org.opensearch.cluster.node.DiscoveryNodeRole
+import org.opensearch.common.settings.ClusterSettings
+import org.opensearch.common.settings.Settings
+import org.opensearch.repositories.RepositoriesService
+import org.opensearch.test.ClusterServiceUtils
+import org.opensearch.test.OpenSearchTestCase
+import org.opensearch.threadpool.TestThreadPool
+import java.util.function.Supplier
+
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
```

Comment:
Let's add test case for removal as well.

Reply:
+1

Reply:
Added cases for removal as well.